readme.md

```diff
@@ -0,0 +1,171 @@
+## Description
+
+This is an SDK for interacting with [CMS Blue Button 2.0 API](https://bluebutton.cms.gov/developers/),
+the API allows applications to obtain a beneficiary's (who has login account with medicare.gov) grant
+to access his/her medicare claims data - through OAUTH2 [(RFC 6749)](https://datatracker.ietf.org/doc/html/rfc6749) authorization flow.
+
+By using the SDK, the development of applications accessing Blue Button 2.0 API can be greatly simplified.
+
+Note, following the OAUTH2 best practices, OAUTH2 PKCE etension [(RFC 7636)](https://datatracker.ietf.org/doc/html/rfc7636) is always enabled.
+
+## Installation
+
+Using npm:
+
+```bash
+$ npm install cms-bluebutton-sdk
+```
+
+Using yarn:
+
+```bash
+$ yarn add cms-bluebutton-sdk
+```
+
+## Configuration
+
+the SDK needs to be properly configured to work, the parameters are:
+
+- the app's credentials - client id, client secret
+- the app's callback url
+- the version number of the API
+- the app's environment (web location where the app is registered)
+
+the configuration is in json format and stored in a local file, the default location
+is current working directory with file name: .bluebutton-config.json
+
+A sample configuration json:
+
+```
+{
+  "clientId": "foo",
+  "clientSecret": "bar",
+  "callbackUrl": "https://www.fake.com/",
+  "version": "1",
+  "environment": "PRODUCTION"
+}
+
+```
+
+| parameter    | value                   | Comments                                                                                                     |
+| ------------ | ----------------------- | ------------------------------------------------------------------------------------------------------------ |
+| clientId     | "foo"                   | oauth2 client id of the app                                                                                  |
```

Comment:
Could maybe use some kind of reference on where to get these client id/secret. Maybe a link out to the sandbox account creation page?